### PR TITLE
[MIG] project_hr: Migration to v12

### PR DESCRIPTION
**Pull request by martintorresvm**

### CODE REVIEW

File: project_hr/README.rst [hunk 1]
```diff
@@ -53,13 +53,13 @@ Configuration
 
 For setting employee categories at project level:
 
-#. Go to *Project > Dashboard*.
-#. Click on the 3 vertical dots for unfolding options and select "Settings".
+#. Go to *Project > Configuration > Settings*.
```
Comment (pedrobaeza) [inline project_hr/README.rst]: This is not correct, and previous steps were correct. The only changed thing is that instead of `Dashboard`, main menu is now called `Projects` (*Project > Projects* in summary)

File: project_hr/README.rst [hunk 2]
```diff
@@ -53,13 +53,13 @@ Configuration
 
 For setting employee categories at project level:
 
-#. Go to *Project > Dashboard*.
-#. Click on the 3 vertical dots for unfolding options and select "Settings".
```
Comment (pedrobaeza) [inline project_hr/README.rst]: If not specifically enough, rephrase it to (also changing the menu name):
```
#. Click on the 3 vertical dots of one of the project kanban cards for unfolding options and select "Edit".
```

File: project_hr/static/description/index.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
-<meta name="generator" content="Docutils 0.15.1: http://docutils.sourceforge.net/" />
+<meta name="generator" content="Docutils 0.14: http://docutils.sourceforge.net/" />
```
Comment (pedrobaeza) [inline project_hr/static/description/index.html]: Regenerate it with new library

File: project_hr/tests/test_project_hr.py [hunk 4]
```diff
@@ -59,6 +59,7 @@ def test_task(self):
             'user_id': self.user_2.id,
             'category_ids': [(6, 0, self.hr_category.ids)],
         })
+        self.task._compute_allowed_user_ids()
```
Comment (pedrobaeza) [inline project_hr/tests/test_project_hr.py]: This mustn't be called manually. If you need to do this, something in the depends is not correct. Other option is that certain operations must call a cache invalidation.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by martintorresvm** [comment]
Changes done @pedrobaeza

File: project_hr/tests/test_project_hr.py [hunk 5]
```diff
@@ -59,6 +59,7 @@ def test_task(self):
             'user_id': self.user_2.id,
             'category_ids': [(6, 0, self.hr_category.ids)],
         })
+        self.task.invalidate_cache()
```
Comment (pedrobaeza) [inline project_hr/tests/test_project_hr.py]: You are invalidating cache here, but you should do it when creating the employee on regular code. If not, on a production database, computed field is not going to be refreshed.

**Comment by martintorresvm** [comment]
Changes done @pedrobaeza

File: project_hr/models/hr_employee.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    @api.model
+    def create(self, vals):
+        employee = super().create(vals)
+        if employee.category_ids:
+            tasks = self.env['project.task'].search([
```
Comment (pedrobaeza) [inline project_hr/models/hr_employee.py]: No need of searching involved tasks. `invalidate_cache` works at model level (it's `@api.model`).

Comment (pedrobaeza) [reply project_hr/models/hr_employee.py]: And what happens when you modify category_ids of an employee? Add a test to check if you need to invalidate cache as well.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: project_hr/models/hr_employee.py (hunk 6 above)
Comment (martintorresvm) [reply project_hr/models/hr_employee.py]: Think already have https://github.com/OCA/project/pull/568/files?file-filters%5B%5D=.py#diff-21c8ed16cfa23508bf379ef5143fdc4eR45

**Comment by martintorresvm** [comment]
Changes done @pedrobaeza

File: project_hr/models/hr_employee.py (hunk 6 above)
Comment (pedrobaeza) [reply project_hr/models/hr_employee.py]: No, do it a second time on same test and it will be then the proper test, as first time there's no value cached.

**Comment by martintorresvm** [comment]
Changes done @pedrobaeza

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@ernestotejeda please review

**Review by ernestotejeda — APPROVED** [review]
Functionally OK
Code OK

**Comment by rafaelbn** [comment]
ocabot /merge
